examples/plot_metadata_routing.py

```diff
@@ -0,0 +1,616 @@
+    def __init__(self, estimator):
+        self.estimator = estimator
+
+    def get_metadata_routing(self):
+        # This method defines the routing for this meta-estimator.
+        # In order to do so, a `MetadataRouter` instance is created, and the
+        # right routing is added to it. More explanation follows.
+        router = MetadataRouter(owner=self.__class__.__name__).add(
+            estimator=self.estimator, method_mapping="one-to-one"
+        )
+        return router
+
+    def fit(self, X, y, **fit_params):
+        # meta-estimators are responsible for validating the given metadata.
+        # `get_routing_for_object` is a safe way to construct a
+        # `MetadataRouter` or a `MetadataRequest` from the given object.
+        request_router = get_routing_for_object(self)
+        request_router.validate_metadata(params=fit_params, method="fit")
+        # we can use provided utility methods to map the given metadata to what
+        # is required by the underlying estimator. Here `method` refers to the
+        # parent's method, i.e. `fit` in this example.
+        routed_params = request_router.route_params(params=fit_params, caller="fit")
+
+        # the output has a key for each object's method which is used here,
+        # i.e. parent's `fit` method, containing the metadata which should be
+        # routed to them, based on the information provided in
+        # ``get_metadata_routing``.
+        self.estimator_ = clone(self.estimator).fit(X, y, **routed_params.estimator.fit)
+        self.classes_ = self.estimator_.classes_
+        return self
+
+    def predict(self, X, **predict_params):
+        check_is_fitted(self)
+        # same as in `fit`, we validate the given metadata
+        request_router = get_routing_for_object(self)
+        request_router.validate_metadata(params=predict_params, method="predict")
+        # and then prepare the input to the underlying `predict` method.
+        params = request_router.route_params(params=predict_params, caller="predict")
+        return self.estimator_.predict(X, **params.estimator.predict)
+
+
+# %%
+# Let's break down different parts of the above code.
+#
+# First, the :meth:`~utils.metadata_routing.get_routing_for_object` takes
+# an object from which a :class:`~utils.metadata_routing.MetadataRouter` or a
+# :class:`~utils.metadata_routing.MetadataRequest` can be constructed. This
+# may be an estimator, or a dictionary representing a ``MetadataRequest`` or
+# ``MetadataRouter`` object. If an estimator is given, it tries to call the
+# estimator's ``get_metadata_routing`` and construct the object from that, and
+# if the estimator doesn't have such a method, then a default empty
+# ``MetadataRequest`` is returned.
+#
+# Then in each method, we use the ``route_params`` method to construct a
+# dictionary of the form ``{"object_name": {"method_name": {"metadata":
+# value}}}`` to pass to the underlying estimator's method.
```

Comment:
Could you add here the explanation for our `routed_params.estimator.fit`? In particular that the middle part `estimator` was caused by `add`ing it to `MetadataRouter`.